service/remotecfg/remotecfg.go

```diff
@@ -118,12 +133,93 @@ var _ service.Service = (*Service)(nil)
 func (s *Service) Run(ctx context.Context, host service.Host) error {
 	s.ctrl = host.NewController(ServiceName)
 
+	// Run the service's own controller.
 	go func() {
 		s.ctrl.Run(ctx)
 	}()
+
+	// We're on the initial start-up of the service.
```

Comment:
I'm not a huge fan of how this always executes, even if the user hasn't configured the service. It feels like we need to refactor this to account for the service running when the user hasn't configured it.

Reply:
Yes, I've found this is buggy. Similarly, there's a race condition depending on whether Update has had a chance to be called before this.

I'll refactor and synchronize this properly.

Reply:
This was a little confusing since we both want to execute the fetching logic on an Update _and_ on the startup of `Run`, but the conditions to it _also_ come from Update.

As a tradeoff, I've kept the initial call here, but added an early exit on `fetch()` so it shouldn't have any effect when the service is not running.